### DISCUSSION TITLE
Tooltip delay when updating a state.

### DISCUSSION THREAD

**Discussion by RafaelCENG** [wuba/react-native-echarts#106] [Q&A] [answered]
As you can see in the video the first part is when im updating the state. What im trying to do is having different backgroundColor in my tooltip based on my index. (doesn't work even with the state)
In the first part you can see that my tooltip is delayed. In the second part i comment out the setter and is perfectly fine.
Any tips about this?
 
 ```js
 tooltip: {
        trigger: "axis",
        formatter: function (params) {
          // setCurrentIndex(params[0].dataIndex);
          return params[0].dataIndex === 0
            ? `${rangeLow.format("DD")}-${rangeLow.add(firstWeek.length-1, "day").format("DD MMM")} \n ${params[0].value.toFixed(1)}%` //prettier-ignore
            : params[0].dataIndex === 52
            ? `${rangeHigh.subtract(lastWeek.length-1, "day").format("DD")}-${rangeHigh.format("DD MMM")} \n ${params[0].value.toFixed(1)}%` //prettier-ignore
            : `${newRangeLow.add(params[0].dataIndex-1, "week").format("DD")}-${newRangeLow.add(params[0].dataIndex-1,"week").add(6,'day').format("DD MMM")} \n ${params[0].value.toFixed(1)}%` //prettier-ignore
        },
        // backgroundColor: currentIndex > 39 ? "#FF5630" : "#A19D9A",
```

https://github.com/wuba/react-native-echarts/assets/45431116/b976440f-c4d6-4d04-a5cc-7580932f81a7

**Comment by RafaelCENG** [comment]
Any news about this one?

**Comment by zhiqingchen** [comment]
This feature is currently not supported in echarts.

**Comment by zhiqingchen** [comment]
If you use useState it will cause re-rendering, which will be delayed.

**Comment by RafaelCENG** [comment]
Is there an other way to make it happen or is not possible currently?

**Reply by zhiqingchen** [reply, marked as answer]
It is possible to make a feature request to the echarts project to have the backgroundColor support callback functions, but this is not a common feature

**Comment by zhiqingchen** [comment]
It is also possible to implement it yourself by patching the echarts package
